src/libs/Navigation/Navigation.js

```diff
@@ -83,10 +83,22 @@ function getDefaultDrawerState(isSmallScreenWidth) {
         return CONST.DRAWER_STATUS.CLOSED;
     }
 
-    // When the user opens the app via Report url, close the drawer else open the drawer.
     const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config).substring(1);
+    const {name, params} = navigationRef.current.getCurrentRoute();
+
+    // Details and IOU screens (with reportID) are open from report screen.
+    // For that we want to correctly order the stack of screens.
     // eslint-disable-next-line no-use-before-define
-    return path === ROUTES.HOME || !isDrawerRoute(path) ? CONST.DRAWER_STATUS.OPEN : CONST.DRAWER_STATUS.CLOSED;
+    if (isDrawerRoute(path)
+      || name === 'Details_Root'
+      || (name === 'IOU_Request_Root' && params.reportID)
+      || (name === 'IOU_Send_Root' && params.reportID)
+      || (name === 'IOU_Bill_Root' && params.reportID)
```

Comment:
@parasharrajat  The details and IOU screens are the screens that can be opened from the report screen, CMIIW.

Reply:
This is not a good solution to hard code screen names.

Reply:
We can move those screen names to constant file, but what do you think about the approach?

Reply:
I already said above not a good approach.

Reply:
Any suggestion then?

So either detail or IOU screens are open from report screen should closed the drawer when reload right? If so, any idea to know what screen is open from the report screen? As far as I know, we do not nest detail or IOU screens to report screen.

Reply:
There are two types of navigation in the APP. 
LHN => Main Drawer
RHN => Any modals that open on the right. They open over the drawer and render as Modal. 

Let's assume that All RHN modals are opened from the LHN screen, then the drawer should be open when RHN modals are directly opened. 

For other navigations, set the correct drawer status based on url.